### ISSUE TITLE
[indent] conflicts with "react/jsx-indent-props"

### ISSUE THREAD

**Issue by tuuling** [typescript-eslint/typescript-eslint#415]
**Repro**

Any JSX code with indented properties

```JSON
{
  "rules": {
    "indent": "off",
    "@typescript-eslint/indent": ["error", 2],
    "react/jsx-indent-props": [1, "first"]
  }
}
```

```TS
 return (<Rhombus key={key}
                  pos-x={tile.cord.pixel.x}
                  pos-y={tile.cord.pixel.y}
                  hover={!!this.state.selectedCord && this.state.selectedCord.key === key}/>)
```

**Expected Result**

No lint error

**Actual Result**

Lint error 

![Screenshot 2019-04-08 at 10 55 38](https://user-images.githubusercontent.com/2890729/55707781-15d65500-59ed-11e9-949b-8cc57313c0b0.png)

**Additional Info**

It seems to be conflicting with https://github.com/yannickcr/eslint-plugin-react/blob/master/docs/rules/jsx-indent-props.md#rule-options when using the "first" option. 

Not sure how this could be fixed. Perhaps an option to ignore jsx indentation and let the react eslint plugin handle that?

**Versions**

| package                            | version |
| ---------------------------------- | ------- |
| `@typescript-eslint/eslint-plugin` | `1.6.0` |
| `@typescript-eslint/parser`        | `1.6.0` |
| `eslint-plugin-react`                    | `7.12.4` |
| `TypeScript`                       | `3.2.4` |
| `ESLint`                           | `5.12.0` |
| `node`                             | `11.8.0` |
| `npm`                              | `6.9.0` |

**Comment by tuuling** [comment]
Right, the solution always come after you make a ticket.

I stumbled upon https://github.com/airbnb/javascript/issues/1569 this ticket. And it seems that using: 
`
"@typescript-eslint/indent": ["error", 2, { "ignoredNodes": ["JSXElement *", "JSXElement"]}]
`
Fixed the problem I had.

I would propose to add an example under "ignoredNodes" selectors section, but the link seems to be broken atm. packages/eslint-plugin/docs/rules/indent.md:93

**Comment by Deliaz** [comment]
No need to disable JSXElelemnt completely. This code should also fix (disables indentation check for attrs only): 
```
"ignoredNodes": [
    "JSXAttribute"
]
``` 
But It will keep validation for the Nodes.

**Comment by pgross41** [comment]
> No need to disable JSXElelemnt completely. This code should also fix (disables indentation check for attrs only):
> 
> ```
> "ignoredNodes": [
>     "JSXAttribute"
> ]
> ```
> 
> But It will keep validation for the Nodes.

If you use the spread operator you will also want to ignore `JSXSpreadAttribute`.
